frontends/p4-14/p4-14-parse.ypp

```diff
@@ -824,31 +824,31 @@ expression:
     | LATEST '.' name   { $$ = new IR::Member(@1+@3, new IR::NamedRef(@1, $1), IR::ID(@3, $3)); }
     | INTEGER           { $$ = $1; }
     | '(' expression ')'                { $$ = $2; }
-    | NOT expression %prec PREFIX       { $$ = new IR::LNot(@1, $2); }
-    | '!' expression %prec PREFIX       { $$ = new IR::LNot(@1, $2); }
-    | '~' expression %prec PREFIX       { $$ = new IR::Cmpl(@1, $2); }
-    | '-' expression %prec PREFIX       { $$ = new IR::Neg(@1, $2); }
+    | NOT expression %prec PREFIX       { $$ = !$2; }
```

Comment:
P4-16 was the same originally, but this causes problems with long expressions that span multiple lines; the error output does not look as sharp - so I moved to pointing just the operator instead of also pointing the operands.